value_test.go

```diff
@@ -551,35 +551,80 @@ func TestValue_IsBoolean(t *testing.T) {
-	data2 := "baz"
-
-	NewValue(reporter, data1).IsEqual(data1).chain.assert(t, success)
-	NewValue(reporter, data2).IsEqual(data2).chain.assert(t, success)
-
-	NewValue(reporter, data1).NotEqual(data1).chain.assert(t, failure)
-	NewValue(reporter, data2).NotEqual(data2).chain.assert(t, failure)
-
-	NewValue(reporter, data1).IsEqual(data2).chain.assert(t, failure)
-	NewValue(reporter, data2).IsEqual(data1).chain.assert(t, failure)
-
-	NewValue(reporter, data1).NotEqual(data2).chain.assert(t, success)
-	NewValue(reporter, data2).NotEqual(data1).chain.assert(t, success)
+	t.Run("basic", func(t *testing.T) {
+		cases := []struct {
+			name         string
+			value1       interface{}
+			value2       interface{}
+			wantEqual    chainResult
+			wantNotEqual chainResult
+		}{
+			{
+				name:         "compare equivalent values (strings)",
+				value1:       "baz",
+				value2:       "baz",
+				wantEqual:    success,
+				wantNotEqual: failure,
+			},
+			{
+				name:         "compare equivalent values (maps)",
+				value1:       map[string]interface{}{"foo": "bar"},
+				value2:       map[string]interface{}{"foo": "bar"},
+				wantEqual:    success,
+				wantNotEqual: failure,
+			},
+			{
+				name:         "compare non-equivalent values",
+				value1:       map[string]interface{}{"foo": "bar"},
+				value2:       "baz",
+				wantEqual:    failure,
+				wantNotEqual: success,
+			},
+			{
+				name:         "compare nil values",
+				value1:       nil,
+				value2:       nil,
+				wantEqual:    success,
+				wantNotEqual: failure,
+			},
+			{
+				name:         "compare nil and nil-value",
+				value1:       nil,
+				value2:       map[string]interface{}(nil),
+				wantEqual:    success,
+				wantNotEqual: failure,
+			},
+			{
+				name:         "compare nil and value",
+				value1:       nil,
+				value2:       map[string]interface{}{},
+				wantEqual:    failure,
+				wantNotEqual: success,
+			},
+			{
+				name:         "compare value and func",
+				value1:       map[string]interface{}{"foo": "bar"},
+				value2:       func() {},
+				wantEqual:    failure,
+				wantNotEqual: failure,
+			},
+		}
 
-	NewValue(reporter, nil).IsEqual(nil).chain.assert(t, success)
+		for _, tc := range cases {
+			t.Run(tc.name, func(t *testing.T) {
+				reporter := newMockReporter(t)
 
-	NewValue(reporter, nil).IsEqual(map[string]interface{}(nil)).chain.assert(t, success)
-	NewValue(reporter, nil).IsEqual(map[string]interface{}{}).chain.assert(t, failure)
+				NewValue(reporter, tc.value1).IsEqual(tc.value2).
+					chain.assert(t, tc.wantEqual)
 
-	NewValue(reporter, data1).IsEqual(func() {}).chain.assert(t, failure)
-	NewValue(reporter, data1).NotEqual(func() {}).chain.assert(t, failure)
+				NewValue(reporter, tc.value1).NotEqual(tc.value2).
+					chain.assert(t, tc.wantNotEqual)
+			})
+		}
+	})
 }
 
 func TestValue_InList(t *testing.T) {
-	reporter := newMockReporter(t)
-
 	data1 := map[string]interface{}{"foo": "bar"}
 	data2 := "baz"
 	data3 := struct {
```

Comment:
What do you think about embedding these values into tests? Most of them are one-line, so it's not much code duplication, but I think tests will be more readable (you don't need to jump to the top and see what is "data1" to understand the test).